# Checkpoint 1: Data Cleaning 1

In [267]:
# Import the numpy and pandas packages

import numpy as np
import pandas as pd

In [268]:
# Supress Warnings

import warnings
warnings.filterwarnings('ignore')

In [269]:
# loading the companies data seperated by tab using latin1 encoding
# After trying different encoding schemes, group found that it works best with latin1
companies = pd.read_csv("companies.txt",delimiter="\t", encoding='latin1')
companies.head()

permalink                    name  \
0                  /Organization/-Fame                   #fame   
1               /Organization/-Qounter                :Qounter   
2  /Organization/-The-One-Of-Them-Inc-  (THE) ONE of THEM,Inc.   
3                /Organization/0-6-Com                 0-6.com   
4       /Organization/004-Technologies        004 Technologies   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
2                http://oneofthem.jp   
3                 http://www.0-6.com   
4  http://004gmbh.de/en/004-interact   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
2                                  Apps|Games|Mobile  operating          NaN   
3                                        Curated Web  operating          CHN   
4                                           Software  operating          USA   

  state_code                 region           city  founded_at  
0         16                 Mumbai         Mumbai         NaN  
1         DE             DE - Other  Delaware City  04-09-2014  
2        NaN                    NaN            NaN         NaN  
3         22                Beijing        Beijing  01-01-2007  
4         IL  Springfield, Illinois      Champaign  01-01-2010

In [270]:
# Manual Inspection for companies dataframe
print("Columns : {}".format(companies.columns))
print("\nShape : {}".format(companies.shape))
print("\nVariable types : \n")
companies.info()
print("\nValues : {}".format(companies.values))
companies.describe()

Columns : Index(['permalink', 'name', 'homepage_url', 'category_list', 'status',
       'country_code', 'state_code', 'region', 'city', 'founded_at'],
      dtype='object')

Shape : (66368, 10)

Variable types : 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 66368 entries, 0 to 66367
Data columns (total 10 columns):
permalink        66368 non-null object
name             66367 non-null object
homepage_url     61310 non-null object
category_list    63220 non-null object
status           66368 non-null object
country_code     59410 non-null object
state_code       57821 non-null object
region           58338 non-null object
city             58340 non-null object
founded_at       51147 non-null object
dtypes: object(10)
memory usage: 5.1+ MB

Values : [['/Organization/-Fame' '#fame' 'http://livfame.com' ... 'Mumbai'
  'Mumbai' nan]
 ['/Organization/-Qounter' ':Qounter' 'http://www.qounter.com' ...
  'DE - Other' 'Delaware City' '04-09-2014']
 ['/Organization/-The-One-Of-Them-Inc-' '(

permalink   name                homepage_url  \
count                   66368  66367                       61310   
unique                  66368  66102                       61191   
top     /Organization/Phytech  Roost  http://www.askforoffer.com   
freq                        1      4                           5   

       category_list     status country_code state_code       region  \
count          63220      66368        59410      57821        58338   
unique         27296          4          137        311         1092   
top         Software  operating          USA         CA  SF Bay Area   
freq            3995      53034        37601      12900         8804   

                 city  founded_at  
count           58340       51147  
unique           5111        3978  
top     San Francisco  01-01-2012  
freq             3526        2730

In [271]:
# loading the rounds2 data seperated by default delimiter i.e. comma using latin1 encoding
# After trying different encoding schemes, group found that it works best with latin1
rounds2 = pd.read_csv("rounds2.csv", encoding="latin1")
rounds2.head()

company_permalink  \
0                  /organization/-fame   
1               /ORGANIZATION/-QOUNTER   
2               /organization/-qounter   
3  /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                /organization/0-6-com   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   

  funding_round_code   funded_at  raised_amount_usd  
0                  B  05-01-2015         10000000.0  
1                  A  14-10-2014                NaN  
2                NaN  01-03-2014           700000.0  
3                  B  30-01-2014          3406878.0  
4                  A  19-03-2008          2000000.0

In [272]:
# Manual Inspection for companies dataframe
print("Columns : {}".format(rounds2.columns))
print("\nShape : {}".format(rounds2.shape))
print("\nVariable types : \n")
rounds2.info()
print("\nValues : {}".format(rounds2.values))
rounds2.describe()

Columns : Index(['company_permalink', 'funding_round_permalink', 'funding_round_type',
       'funding_round_code', 'funded_at', 'raised_amount_usd'],
      dtype='object')

Shape : (114949, 6)

Variable types : 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 114949 entries, 0 to 114948
Data columns (total 6 columns):
company_permalink          114949 non-null object
funding_round_permalink    114949 non-null object
funding_round_type         114949 non-null object
funding_round_code         31140 non-null object
funded_at                  114949 non-null object
raised_amount_usd          94959 non-null float64
dtypes: float64(1), object(5)
memory usage: 5.3+ MB

Values : [['/organization/-fame' '/funding-round/9a01d05418af9f794eebff7ace91f638'
  'venture' 'B' '05-01-2015' 10000000.0]
 ['/ORGANIZATION/-QOUNTER'
  '/funding-round/22dacff496eb7acb2b901dec1dfe5633' 'venture' 'A'
  '14-10-2014' nan]
 ['/organization/-qounter'
  '/funding-round/b44fbb94153f6cdef13083530bb48030' 'seed' n

raised_amount_usd
count       9.495900e+04
mean        1.042687e+07
std         1.148212e+08
min         0.000000e+00
25%         3.225000e+05
50%         1.680511e+06
75%         7.000000e+06
max         2.127194e+10

In [273]:
# It can be observed from both the files headers that files companies and rounds2 are in mixed different case letters.
# To appropriately merge these files, merge column need to converted to lower case for comparison
# Since both the files are taken in 'latin1' encoding, encoding and decoding to other format may not be required.

# converting permalink for companies into lower case
companies['permalink'] = companies['permalink'].str.lower()
companies.head()


permalink                    name  \
0                  /organization/-fame                   #fame   
1               /organization/-qounter                :Qounter   
2  /organization/-the-one-of-them-inc-  (THE) ONE of THEM,Inc.   
3                /organization/0-6-com                 0-6.com   
4       /organization/004-technologies        004 Technologies   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
2                http://oneofthem.jp   
3                 http://www.0-6.com   
4  http://004gmbh.de/en/004-interact   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
2                                  Apps|Games|Mobile  operating          NaN   
3                                        Curated Web  operating          CHN   
4                                           Software  operating          USA   

  state_code                 region           city  founded_at  
0         16                 Mumbai         Mumbai         NaN  
1         DE             DE - Other  Delaware City  04-09-2014  
2        NaN                    NaN            NaN         NaN  
3         22                Beijing        Beijing  01-01-2007  
4         IL  Springfield, Illinois      Champaign  01-01-2010

In [274]:
# It can be observed from both the files headers that files companies and rounds2 are in mixed different case letters.
# To appropriately merge these files, merge column need to converted to lower case for comparison
# Since both the files are taken in 'latin1' encoding, encoding and decoding to other format may not be required.

# converting company_permalink for rounds2 into lower case
rounds2['company_permalink'] = rounds2['company_permalink'].str.lower()
rounds2.head()

company_permalink  \
0                  /organization/-fame   
1               /organization/-qounter   
2               /organization/-qounter   
3  /organization/-the-one-of-them-inc-   
4                /organization/0-6-com   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   

  funding_round_code   funded_at  raised_amount_usd  
0                  B  05-01-2015         10000000.0  
1                  A  14-10-2014                NaN  
2                NaN  01-03-2014           700000.0  
3                  B  30-01-2014          3406878.0  
4                  A  19-03-2008          2000000.0

In [275]:
# Unique companies present in round2
len(set(rounds2['company_permalink']))

66370

In [276]:
# Unique companies present in company files
len(set(companies['permalink']))

66368

In [277]:
#since len(companies['permalink'].unique()) = no of rows in companies data frame
# permalink should be the unique key
len(companies['permalink'].unique())

66368

In [278]:
# also checked that name are not unique across dataframe.
len(companies['name'].unique())

66103


### Merging the companies data with round2 into `master_frame`.


In [279]:
#inner join is based on analysis requirement that is depends on combined data for both companies and rounds2
#from companies data related to country_code and category_list is essential
#from rounds2 to the raised_amount_usd and funding_round_type are improtant
# a new copy for the new merged dataframe is created
master_frame = pd.DataFrame(pd.merge(rounds2,companies, how='inner', 
                                     left_on = 'company_permalink', right_on = 'permalink')).copy()
master_frame.head()

company_permalink  \
0                  /organization/-fame   
1               /organization/-qounter   
2               /organization/-qounter   
3  /organization/-the-one-of-them-inc-   
4                /organization/0-6-com   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   

  funding_round_code   funded_at  raised_amount_usd  \
0                  B  05-01-2015         10000000.0   
1                  A  14-10-2014                NaN   
2                NaN  01-03-2014           700000.0   
3                  B  30-01-2014          3406878.0   
4                  A  19-03-2008          2000000.0   

                             permalink                    name  \
0                  /organization/-fame                   #fame   
1               /organization/-qounter                :Qounter   
2               /organization/-qounter                :Qounter   
3  /organization/-the-one-of-them-inc-  (THE) ONE of THEM,Inc.   
4                /organization/0-6-com                 0-6.com   

             homepage_url                                      category_list  \
0      http://livfame.com                                              Media   
1  http://www.qounter.com  Application Platforms|Real Time|Social Network...   
2  http://www.qounter.com  Application Platforms|Real Time|Social Network...   
3     http://oneofthem.jp                                  Apps|Games|Mobile   
4      http://www.0-6.com                                        Curated Web   

      status country_code state_code      region           city  founded_at  
0  operating          IND         16      Mumbai         Mumbai         NaN  
1  operating          USA         DE  DE - Other  Delaware City  04-09-2014  
2  operating          USA         DE  DE - Other  Delaware City  04-09-2014  
3  operating          NaN        NaN         NaN            NaN         NaN  
4  operating          CHN         22     Beijing        Beijing  01-01-2007

In [280]:
# Manual Inspection for merged master_frame dataframe
print("Columns : {}".format(master_frame.columns))
print("\nShape : {}".format(master_frame.shape))
print("\nVariable types : \n")
master_frame.info()
print("\nValues : {}".format(master_frame.values))
master_frame.describe()

Columns : Index(['company_permalink', 'funding_round_permalink', 'funding_round_type',
       'funding_round_code', 'funded_at', 'raised_amount_usd', 'permalink',
       'name', 'homepage_url', 'category_list', 'status', 'country_code',
       'state_code', 'region', 'city', 'founded_at'],
      dtype='object')

Shape : (114942, 16)

Variable types : 

<class 'pandas.core.frame.DataFrame'>
Int64Index: 114942 entries, 0 to 114941
Data columns (total 16 columns):
company_permalink          114942 non-null object
funding_round_permalink    114942 non-null object
funding_round_type         114942 non-null object
funding_round_code         31139 non-null object
funded_at                  114942 non-null object
raised_amount_usd          94958 non-null float64
permalink                  114942 non-null object
name                       114941 non-null object
homepage_url               108810 non-null object
category_list              111535 non-null object
status                     114942 n

raised_amount_usd
count       9.495800e+04
mean        1.042696e+07
std         1.148218e+08
min         0.000000e+00
25%         3.225000e+05
50%         1.680521e+06
75%         7.000000e+06
max         2.127194e+10

In [281]:
# As you can observe that in above data set there are few companies in rounds2 for which 
# there is no data set in companies data frame.
# the data is in same encoding latin1 for both companies and rounds2
# and contains non-english or special characters in permalink which couldn't be compared efficiently.
# On manual check it was also found that the count for such data is quiet less.
set(rounds2['company_permalink']).difference(set(companies['permalink']))

{'/organization/e-cã\x8abica',
 '/organization/energystone-games-ç\x81µç\x9f³æ¸¸æ\x88\x8f',
 '/organization/huizuche-com-æ\x83\xa0ç§ÿ车',
 '/organization/magnet-tech-ç£\x81ç\x9f³ç§\x91æ\x8a\x80',
 '/organization/tipcat-interactive-æ²\x99è\x88ÿä¿¡æ\x81¯ç§\x91æ\x9a\x80',
 '/organization/weiche-tech-å\x96\x82车ç§\x91æ\x8a\x80',
 '/organization/zengame-ç¦\x85游ç§\x91æ\x8a\x80'}

In [282]:
# similar inspection for companies data not present in rounds2
set(companies['permalink']).difference(set(rounds2['company_permalink']))

{'/organization/e-cã\x9abica',
 '/organization/energystone-games-ç\x81µçÿ³æ¸¸æ\x88\x8f',
 '/organization/huizuche-com-æ\x83\xa0ç§\x9f车',
 '/organization/tipcat-interactive-æ²\x99è\x88\x9fä¿¡æ\x81¯ç§\x91æ\x9a\x80',
 '/organization/zengame-ç¦\x85游ç§\x91æ\x9a\x80'}


### Start data cleaning process for `master_frame`


In [283]:
# check the column-wise null count
master_frame.isnull().sum()

company_permalink              0
funding_round_permalink        0
funding_round_type             0
funding_round_code         83803
funded_at                      0
raised_amount_usd          19984
permalink                      0
name                           1
homepage_url                6132
category_list               3407
status                         0
country_code                8671
state_code                 10939
region                     10160
city                       10157
founded_at                 20520
dtype: int64

In [284]:
# check he percentage of null data in each column of master_frame
print(round(100*(master_frame.isnull().sum()/len(master_frame.index)),2))

company_permalink           0.00
funding_round_permalink     0.00
funding_round_type          0.00
funding_round_code         72.91
funded_at                   0.00
raised_amount_usd          17.39
permalink                   0.00
name                        0.00
homepage_url                5.33
category_list               2.96
status                      0.00
country_code                7.54
state_code                  9.52
region                      8.84
city                        8.84
founded_at                 17.85
dtype: float64


In [285]:
# We are analyzing the investments done by others
# in absence of amount funded it is not relevant to take that data
#so this needs to be removed
# deleting rows with raised_amount_usd is Null
master_frame = master_frame[~np.isnan(master_frame['raised_amount_usd'])]
print(round(100*(master_frame.isnull().sum()/len(master_frame.index)),2))

company_permalink           0.00
funding_round_permalink     0.00
funding_round_type          0.00
funding_round_code         70.34
funded_at                   0.00
raised_amount_usd           0.00
permalink                   0.00
name                        0.00
homepage_url                4.56
category_list               1.10
status                      0.00
country_code                6.16
state_code                  8.01
region                      7.42
city                        7.42
founded_at                 16.81
dtype: float64


In [286]:
# for the investment to be performed the country code sould be known
# as one of the major analysis is based on the country and that too English speaking country
# therefore, the data with null values for country codes makes no sence
# and needs to be deleted from the master_frame
# deleting rows with  country_code is not present
master_frame = master_frame[~pd.isnull(master_frame['country_code'])]
print(round(100*(master_frame.isnull().sum()/len(master_frame.index)),2))

company_permalink           0.00
funding_round_permalink     0.00
funding_round_type          0.00
funding_round_code         69.77
funded_at                   0.00
raised_amount_usd           0.00
permalink                   0.00
name                        0.00
homepage_url                3.79
category_list               0.65
status                      0.00
country_code                0.00
state_code                  1.97
region                      1.34
city                        1.34
founded_at                 15.33
dtype: float64


In [287]:
#For the sector analysis also the data for category_list is required.
# If case this data is missing, investment decision cannot happen
# So deleting the rows from the masterframe where category_list is not present
master_frame = master_frame[~pd.isnull(master_frame['category_list'])]
print(round(100*(master_frame.isnull().sum()/len(master_frame.index)),2))

company_permalink           0.00
funding_round_permalink     0.00
funding_round_type          0.00
funding_round_code         69.75
funded_at                   0.00
raised_amount_usd           0.00
permalink                   0.00
name                        0.00
homepage_url                3.67
category_list               0.00
status                      0.00
country_code                0.00
state_code                  1.96
region                      1.34
city                        1.33
founded_at                 15.10
dtype: float64


In [288]:
# dropping the non-required columns after reviewing what is required across all checkpoints
# company_permalink is dropped as it was duplicate
#funding_round_code,homepage_url,state_code,region,city,founded_at are dropped as not required for analysis
dropping_columns = ['company_permalink','funding_round_code','homepage_url','state_code','region','city','founded_at']
master_frame = master_frame.drop(dropping_columns,axis=1)
print(round(100*(master_frame.isnull().sum()/len(master_frame.index)),2))

funding_round_permalink    0.0
funding_round_type         0.0
funded_at                  0.0
raised_amount_usd          0.0
permalink                  0.0
name                       0.0
category_list              0.0
status                     0.0
country_code               0.0
dtype: float64


In [289]:
# final master_frame stats for null data after data cleaning
# all percentages as 0 suggest that the data is finally cleaned and can be used for apprpriate analysis
print(round(100*(master_frame.isnull().sum()/len(master_frame.index)),2))

funding_round_permalink    0.0
funding_round_type         0.0
funded_at                  0.0
raised_amount_usd          0.0
permalink                  0.0
name                       0.0
category_list              0.0
status                     0.0
country_code               0.0
dtype: float64


In [290]:
# check the master_frame as how many number of rows and percentage of rows that have been retained.
print("number of rows present now : {}".format(len(master_frame.index)))      
percent_rows_retained = round(100*(len(master_frame.index)/114942),2)
print("percent rows retained : {}".format(percent_rows_retained))

number of rows present now : 88529
percent rows retained : 77.02


#### Information : After data cleaning, we have retained 77.02 % of the data in the `master_frame`

In [291]:
#cleaned master_frame now looks like
master_frame

funding_round_permalink funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
2       /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
4       /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
6       /funding-round/7d53696f2b4f607a2f2a8cbb83d01839        undisclosed   
7       /funding-round/2b9d3ac293d5cdccbecff5c8cb0f327d               seed   
8       /funding-round/954b9499724b946ad8c396a57a5f3b72            venture   
9       /funding-round/383a9bd2c04f7038bb543ccef5ba3eae               seed   
10      /funding-round/3bb2ee4a2d89251a10aaa735b1180e44            venture   
11      /funding-round/ae2a174c06517c2394aed45006322a7e            venture   
12      /funding-round/e1cfcbe1bdf4c70277c5f29a3482f24e            venture   
13      /funding-round/03b975068632eba5bfdb937ec8c07a68               seed   
15      /funding-round/e82464f22241715dd1a6c77241055ed1               seed   
16      /funding-round/6de4609e894495105bc791ed64361288               seed   
20      /funding-round/9eb8c7790a0c200d79e75785d1c4aa12   convertible_note   
21      /funding-round/11c228f58831bc7ed337ef69ecc560c2     private_equity   
22      /funding-round/b952cbaf401f310927430c97b68162ea            venture   
23      /funding-round/b5c811f561cfb5701cd5b1762120bbc3   convertible_note   
24      /funding-round/436874dc54b54460f99f44c06e50990a     debt_financing   
25      /funding-round/8025ee9c11a586eac4a234ddec7beb30     debt_financing   
26      /funding-round/a884bf8326b9fd59789c0f74a28dbc3b     debt_financing   
27      /funding-round/d205f5416ce55e126beda3ed2bf2c2cd     debt_financing   
28      /funding-round/0faccbbcc5818dc5326469f13f5a8ac8            venture   
29      /funding-round/f245a74b4c54610ae843e17bdf4d1113               seed   
30      /funding-round/8f4fa70e24269ddd8bdf6658906b356b               seed   
31      /funding-round/ce6749b6441f52ccaf38b6d5578779fd               seed   
34      /funding-round/502bd0e50c27616995e4bdad24605ef8            venture   
35      /funding-round/f5fb428ed1e63321f88b3e646ba85f00               seed   
36      /funding-round/01c1aa44619ae45a16e3098fc73facfa     debt_financing   
37      /funding-round/0953b01a028ac4cd075415554cb363b4     debt_financing   
38      /funding-round/13be128d655076a025221d7fddc90d68            venture   
...                                                 ...                ...   
114909  /funding-round/05f43b37c270f5919578281ce803de8f            venture   
114910  /funding-round/129127802dd48be42d6cd45f65895245            venture   
114911  /funding-round/6c01f0a76c1341118d63b670ac841bd9               seed   
114912  /funding-round/dee233b3434185c14b3352034f72d798              angel   
114913  /funding-round/4191cb5258d3863b0212d1083881da64            venture   
114914  /funding-round/85443f5438bfd50104b9591eabc95c94              grant   
114915  /funding-round/b4fce7f8a13267353d57db09099800e8            venture   
114916  /funding-round/ea5b21da6cf999ff4d5667431fe318d7            venture   
114917  /funding-round/24e8134882bf6c4dcc385abba0acd5d9               seed   
114918  /funding-round/c9f5097b3617c7539aa47c8b448163be               seed   
114919  /funding-round/472c6f7940522552fcca465b8cd3deef               seed   
114921  /funding-round/3c3ae0f177e01a04deea811e1828be35     private_equity   
114922  /funding-round/55f33b179dd6f4516315348c1e8350c6              angel   
114923  /funding-round/c05aab2155e390066d51c6852ff61464              grant   
114924  /funding-round/e949f077c943db197f20e43c2eb8d0e6            venture   
114925  /funding-round/2a13fa52848ace2760b224a350a76b40               seed   
114926  /funding-round/39baf35e59eb3efc5409cd4c5eaa50e1               seed   
114927  /funding-round/7a0fa791c96c9c447ee54662395beab9            venture   
114928  /funding-round/cc8cd96a500332d5b4f4bcbddb306099              angel   
114929  /funding-round/7ea37c0ff4132b084832039a9380d34

Checkpoint 1 finishes here and now we have `master_frame` cleaned up for further analysis

# Checkpoint 2: Funding Type Analysis

In [292]:
# As part of Funding type analysis we are required to compare the amounts in million usd.
# however the amounts mentioned in raised_amount_usd column is in usd and mentioned in exp. form
# this is dufficult to compare
# therefore, a new column raised_amount_million_usd is added to master_frame that contains million dollars
# also the decimal places are taken till 3 as it is based on international counting standard
master_frame['raised_amount_million_usd'] = master_frame['raised_amount_usd'].apply(lambda x: round(x/1000000,3))
master_frame.head()

funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
6  /funding-round/7d53696f2b4f607a2f2a8cbb83d01839        undisclosed   
7  /funding-round/2b9d3ac293d5cdccbecff5c8cb0f327d               seed   

    funded_at  raised_amount_usd                            permalink  \
0  05-01-2015         10000000.0                  /organization/-fame   
2  01-03-2014           700000.0               /organization/-qounter   
4  19-03-2008          2000000.0                /organization/0-6-com   
6  01-07-2014            41250.0     /organization/01games-technology   
7  11-09-2009            43360.0  /organization/0ndine-biomedical-inc   

                     name                                      category_list  \
0                   #fame                                              Media   
2                :Qounter  Application Platforms|Real Time|Social Network...   
4                 0-6.com                                        Curated Web   
6      01Games Technology                                              Games   
7  Ondine Biomedical Inc.                                      Biotechnology   

      status country_code  raised_amount_million_usd  
0  operating          IND                     10.000  
2  operating          USA                      0.700  
4  operating          CHN                      2.000  
6  operating          HKG                      0.041  
7  operating          CAN                      0.043

In [293]:
# as per requirement we are grouping the master_frame based on funding_round_type
# on the group, we are applying the mean aggrgated function
# to get the mean values in raised_amount_million_usd
master_frame_groupby = master_frame.groupby('funding_round_type')
master_frame_groupby.mean()

raised_amount_usd  raised_amount_million_usd
funding_round_type                                                 
angel                       9.715739e+05                   0.971575
convertible_note            1.337187e+06                   1.337184
debt_financing              1.716765e+07                  17.167651
equity_crowdfunding         5.098980e+05                   0.509903
grant                       4.512698e+06                   4.512693
non_equity_assistance       4.807534e+05                   0.480767
post_ipo_debt               1.694518e+08                 169.451801
post_ipo_equity             6.607706e+07                  66.077052
private_equity              7.393849e+07                  73.938484
product_crowdfunding        1.353227e+06                   1.353221
secondary_market            8.443853e+07                  84.438536
seed                        7.477937e+05                   0.747793
undisclosed                 1.589166e+07                  15.891662
venture                     1.172422e+07                  11.724223

### Conclusion : On checking above result, it is observed that between 5 to 15 million USD there is only one funding round type which is for `venture` funding type

In [294]:
# creating the dataframe copy for the most applicable funding_round_type "venture"
# which can be used further in later analsis
venture_frame = pd.DataFrame(master_frame.loc[master_frame.funding_round_type=='venture']).copy()
venture_frame.head()

funding_round_permalink funding_round_type  \
0   /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
4   /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
8   /funding-round/954b9499724b946ad8c396a57a5f3b72            venture   
10  /funding-round/3bb2ee4a2d89251a10aaa735b1180e44            venture   
11  /funding-round/ae2a174c06517c2394aed45006322a7e            venture   

     funded_at  raised_amount_usd                            permalink  \
0   05-01-2015         10000000.0                  /organization/-fame   
4   19-03-2008          2000000.0                /organization/0-6-com   
8   21-12-2009           719491.0  /organization/0ndine-biomedical-inc   
10  09-11-2015         20000000.0                 /organization/0xdata   
11  03-01-2013          1700000.0                 /organization/0xdata   

                      name  category_list     status country_code  \
0                    #fame          Media  operating          IND   
4                  0-6.com    Curated Web  operating          CHN   
8   Ondine Biomedical Inc.  Biotechnology  operating          CAN   
10                  H2O.ai      Analytics  operating          USA   
11                  H2O.ai      Analytics  operating          USA   

    raised_amount_million_usd  
0                      10.000  
4                       2.000  
8                       0.719  
10                     20.000  
11                      1.700


# Checkpoint 3: Country Analysis


In [295]:
# As required, this analysis is further derived after the results derved from checkpoint 2 
# wherein we concluded that "venture" investment is most applicable investment
# Now, we are further grouping the data for venture_frame based on country_code
# we use this grouping to evaluate the sum of raised_amount_million_usd across top
# 9 countries where the investment is maximum.
venture_frame_groupby = venture_frame.groupby('country_code')
top9 = pd.DataFrame(venture_frame_groupby.sum()
                    .sort_values(by='raised_amount_million_usd', ascending = False).head(9)).copy()
top9

raised_amount_usd  raised_amount_million_usd
country_code                                              
USA                4.200680e+11                 420068.059
CHN                3.933892e+10                  39338.918
GBR                2.007281e+10                  20072.815
IND                1.426151e+10                  14261.509
CAN                9.482218e+09                   9482.208
FRA                7.226851e+09                   7226.847
ISR                6.854350e+09                   6854.350
DEU                6.306922e+09                   6306.922
JPN                3.167647e+09                   3167.647

 ### Conclusion : After doing manual check for country codes, to find top three english speaking countries among above, we have following results
- #### Top English speaking country - United States
- #### Second English speaking country - United Kingdom
- #### Third English speaking country - India


# Checkpoint 4: Sector Analysis 1


In [296]:
#After reviewing the data we can see that the funding is happening across a lot of sectors
#Lets divide them into 8 main sectors to make our task easy
#For that, first we need to know which one is the primary sector
# Extracting primary sector among category list into a new column Primary_Sector
master_frame['Primary_Sector'] = master_frame['category_list'].str.split("|", expand = True)[0]

#Check to make sure that values are populated correctly
master_frame[['category_list','Primary_Sector']].head()

category_list         Primary_Sector
0                                              Media                  Media
2  Application Platforms|Real Time|Social Network...  Application Platforms
4                                        Curated Web            Curated Web
6                                              Games                  Games
7                                      Biotechnology          Biotechnology

In [297]:
#Now, we know the primary sector. Lets find out which is the main sector.
# For that, lets first import the data of mapping file.
mapping = pd.read_csv("mapping.csv", encoding='latin1')
mapping.head()

category_list  Automotive & Sports  Blanks  Cleantech / Semiconductors  \
0            NaN                    0       1                           0   
1             3D                    0       0                           0   
2    3D Printing                    0       0                           0   
3  3D Technology                    0       0                           0   
4     Accounting                    0       0                           0   

   Entertainment  Health  Manufacturing  News, Search and Messaging  Others  \
0              0       0              0                           0       0   
1              0       0              1                           0       0   
2              0       0              1                           0       0   
3              0       0              1                           0       0   
4              0       0              0                           0       0   

   Social, Finance, Analytics, Advertising  
0                                        0  
1                                        0  
2                                        0  
3                                        0  
4                                        1

In [298]:
# Manual Inspection for merged mapping dataframe
print("Columns : {}".format(mapping.columns))
print("\nShape : {}".format(mapping.shape))
print("\nVariable types : \n")
mapping.info()
print("\nValues : {}".format(mapping.values))
mapping.describe()

Columns : Index(['category_list', 'Automotive & Sports', 'Blanks',
       'Cleantech / Semiconductors', 'Entertainment', 'Health',
       'Manufacturing', 'News, Search and Messaging', 'Others',
       'Social, Finance, Analytics, Advertising'],
      dtype='object')

Shape : (688, 10)

Variable types : 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 688 entries, 0 to 687
Data columns (total 10 columns):
category_list                              687 non-null object
Automotive & Sports                        688 non-null int64
Blanks                                     688 non-null int64
Cleantech / Semiconductors                 688 non-null int64
Entertainment                              688 non-null int64
Health                                     688 non-null int64
Manufacturing                              688 non-null int64
News, Search and Messaging                 688 non-null int64
Others                                     688 non-null int64
Social, Finance, Analytics, A

Automotive & Sports      Blanks  Cleantech / Semiconductors  \
count           688.000000  688.000000                  688.000000   
mean              0.031977    0.001453                    0.077035   
std               0.176066    0.038125                    0.266841   
min               0.000000    0.000000                    0.000000   
25%               0.000000    0.000000                    0.000000   
50%               0.000000    0.000000                    0.000000   
75%               0.000000    0.000000                    0.000000   
max               1.000000    1.000000                    1.000000   

       Entertainment      Health  Manufacturing  News, Search and Messaging  \
count     688.000000  688.000000     688.000000                  688.000000   
mean        0.129360    0.091570       0.058140                    0.104651   
std         0.335843    0.288627       0.234177                    0.306326   
min         0.000000    0.000000       0.000000                    0.000000   
25%         0.000000    0.000000       0.000000                    0.000000   
50%         0.000000    0.000000       0.000000                    0.000000   
75%         0.000000    0.000000       0.000000                    0.000000   
max         1.000000    1.000000       1.000000                    1.000000   

           Others  Social, Finance, Analytics, Advertising  
count  688.000000                               688.000000  
mean     0.283430                                 0.222384  
std      0.450991                                 0.416150  
min      0.000000                                 0.000000  
25%      0.000000                                 0.000000  
50%      0.000000                                 0.000000  
75%      1.000000                                 0.000000  
max      1.000000                                 1.000000

In [299]:
#On manual analysis, it was noticed that at some places instead of word na 0 is written like Analytics 
# is written as A0lytics. Lets change it first. Otherwise we wont be able to recode all the sectors
mapping['category_list'].replace(to_replace = '0',value = 'na', inplace = True,  regex=True)
#Execution of above statement also changes Enterprise 2.0 to Enterprise 2.na which is not needed. 
#So, lets change it back to its original value.
mapping['category_list'].replace(to_replace = 'Enterprise 2.na',value = 'Enterprise 2.0', inplace = True,  regex=True)
#Check if the recoding is done properly. You can see that now A0lytics is indeed Analytics
mapping['category_list'].head(50)

0                                    NaN
1                                     3D
2                            3D Printing
3                          3D Technology
4                             Accounting
5                       Active Lifestyle
6                           Ad Targeting
7                     Advanced Materials
8                       Adventure Travel
9                            Advertising
10                 Advertising Exchanges
11                  Advertising Networks
12                 Advertising Platforms
13                                Advice
14                             Aerospace
15                           Agriculture
16                 Air Pollution Control
17                            Algorithms
18                           All Markets
19                          All Students
20                  Alternative Medicine
21                                Alumni
22                             Analytics
23                               Android
24              

In [300]:
# It is observed that the mapping dataframe is not straight forward to extract the main sectors.
# In order to extract the main sectors we need to prepare a seperate dataframe that provides the
# apropriate mapping for category_list and main_sectors.
# This new dataframe called melted_dataframe is prepared by using melt function.
# the result dataframe contains category_list, main_sector 
# and sec_value(0 or 1 as mentioned in original mapping dataframe).
cols =  list(mapping.columns)
cols.remove('category_list')
#To make it simple lets first change the data from wide format to long format using melt function
melted_mapping = mapping.melt(id_vars = 'category_list', 
                        value_vars = cols, var_name='main_sector', value_name = 'sec_value')
melted_mapping.head(20)

category_list          main_sector  sec_value
0                     NaN  Automotive & Sports          0
1                      3D  Automotive & Sports          0
2             3D Printing  Automotive & Sports          0
3           3D Technology  Automotive & Sports          0
4              Accounting  Automotive & Sports          0
5        Active Lifestyle  Automotive & Sports          0
6            Ad Targeting  Automotive & Sports          0
7      Advanced Materials  Automotive & Sports          0
8        Adventure Travel  Automotive & Sports          1
9             Advertising  Automotive & Sports          0
10  Advertising Exchanges  Automotive & Sports          0
11   Advertising Networks  Automotive & Sports          0
12  Advertising Platforms  Automotive & Sports          0
13                 Advice  Automotive & Sports          0
14              Aerospace  Automotive & Sports          1
15            Agriculture  Automotive & Sports          0
16  Air Pollution Control  Automotive & Sports          0
17             Algorithms  Automotive & Sports          0
18            All Markets  Automotive & Sports          0
19           All Students  Automotive & Sports          0

In [301]:
# among the melted_mapping we require only the mappings where the sec_value is 1
# a value of sec_value=0 means that the category_list doesn't belong to the main_category
# after filtering the required values we shall be left with appropriate dataframe
# containing strainght forward and useable mappings for category_list and main_sector
melted_mapping = melted_mapping[melted_mapping['sec_value']==1]
melted_mapping.head(50)

category_list                 main_sector  sec_value
8             Adventure Travel         Automotive & Sports          1
14                   Aerospace         Automotive & Sports          1
45                        Auto         Automotive & Sports          1
46             Automated Kiosk         Automotive & Sports          1
47                  Automotive         Automotive & Sports          1
57                    Bicycles         Automotive & Sports          1
69            Boating Industry         Automotive & Sports          1
87                         CAD         Automotive & Sports          1
93                        Cars         Automotive & Sports          1
188                     Design         Automotive & Sports          1
189                  Designers         Automotive & Sports          1
213                     Drones         Automotive & Sports          1
226          Electric Vehicles         Automotive & Sports          1
292                      Fuels         Automotive & Sports          1
331             Heavy Industry         Automotive & Sports          1
364      Industrial Automation         Automotive & Sports          1
409                 Limousines         Automotive & Sports          1
602                     Soccer         Automotive & Sports          1
626             Sporting Goods         Automotive & Sports          1
627                     Sports         Automotive & Sports          1
652             Transportation         Automotive & Sports          1
657       Unmanned Air Systems         Automotive & Sports          1
688                        NaN                      Blanks          1
1392     Air Pollution Control  Cleantech / Semiconductors          1
1409               Aquaculture  Cleantech / Semiconductors          1
1437                 Bio-Pharm  Cleantech / Semiconductors          1
1438                  Biofuels  Cleantech / Semiconductors          1
1439            Bioinformatics  Cleantech / Semiconductors          1
1440  Biomass Power Generation  Cleantech / Semiconductors          1
1441                Biometrics  Cleantech / Semiconductors          1
1442             Biotechnology  Cleantech / Semiconductors          1
1443                   Bitcoin  Cleantech / Semiconductors          1
1466                    Carbon  Cleantech / Semiconductors          1
1477                 Chemicals  Cleantech / Semiconductors          1
1482              Clean Energy  Cleantech / Semiconductors          1
1483          Clean Technology  Cleantech / Semiconductors          1
1484       Clean Technology IT  Cleantech / Semiconductors          1
1501          Commercial Solar  Cleantech / Semiconductors          1
1502               Commodities  Cleantech / Semiconductors          1
1509  Concentrated Solar Power  Cleantech / Semiconductors          1
1581         Disruptive Models  Cleantech / Semiconductors          1
1593    Early-Stage Technology  Cleantech / Semiconductors          1
1613                    Energy  Cleantech / Semiconductors          1
1614         Energy Efficiency  Cleantech / Semiconductors          1
1615                 Energy IT  Cleantech / Semiconductors          1
1616         Energy Management  Cleantech / Semiconductors          1
1617            Energy Storage  Cleantech / Semiconductors          1
1630  Environmental Innovation  Cleantech / Semiconductors          1
1643                 Fertility  Cleantech / Semiconductors          1
1667                Fuel Cells  Cleantech / Semiconductors          1

In [302]:
#Perform the merging of master_frame and mapping database to populate the main_sector in master_frame
# inner join is taken so that blanks values which cannot be mapped to any of  
# the main sectors in mapping table do not appear in the result
master_frame = pd.merge(master_frame, melted_mapping[['main_sector']], 
                        how='inner', 
                        left_on = master_frame.Primary_Sector.str.lower(), 
                        right_on = melted_mapping.category_list.str.lower())

#Check the data after merging to make sure it is populated correctly
master_frame.head()

funding_round_permalink   funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638              venture   
1  /funding-round/21a2cbf6f2fb2a1c2a61e04bf930dfe6              venture   
2  /funding-round/bd626ed022f5c66574b1afe234f3c90d              venture   
3  /funding-round/fd4b15e8c97ee2ffc0acccdbe1a98810              venture   
4  /funding-round/9ab9dbd17bf010c79d8415b2c22be6fa  equity_crowdfunding   

    funded_at  raised_amount_usd                     permalink  \
0  05-01-2015         10000000.0           /organization/-fame   
1  06-10-2015         15000000.0           /organization/90min   
2  07-05-2013          5800000.0           /organization/90min   
3  26-03-2014         18000000.0           /organization/90min   
4  26-03-2014          1090000.0  /organization/a-dance-for-me   

             name                        category_list     status  \
0           #fame                                Media  operating   
1           90min  Media|News|Publishing|Soccer|Sports  operating   
2           90min  Media|News|Publishing|Soccer|Sports  operating   
3           90min  Media|News|Publishing|Soccer|Sports  operating   
4  A Dance for Me       Media|News|Photo Sharing|Video  operating   

  country_code  raised_amount_million_usd Primary_Sector    main_sector  
0          IND                      10.00          Media  Entertainment  
1          GBR                      15.00          Media  Entertainment  
2          GBR                       5.80          Media  Entertainment  
3          GBR                      18.00          Media  Entertainment  
4          USA                       1.09          Media  Entertainment

# Checkpoint 5: Sector Analysis 2

In [303]:
# creating new dataframe D1 for country United States (USA)
# the data is extracted from master_frame that contains
# all the initially required columns
D1 = pd.DataFrame(master_frame[(master_frame.funding_round_type == 'venture') 
                               & (master_frame.country_code == 'USA') 
                               & (master_frame.raised_amount_million_usd >= 5) 
                               & (master_frame.raised_amount_million_usd <= 15)
                              ]
                 ).copy()
D1.head()

funding_round_permalink funding_round_type  \
7   /funding-round/452a2342fe720285c3b92e9bd927d9ba            venture   
31  /funding-round/adca195749ae9ace84684723fbe75e5b            venture   
61  /funding-round/7f05940c4d2dfecb8e50a0e5720e5065            venture   
62  /funding-round/9241ae16e08df17ebdc064e49e23035a            venture   
85  /funding-round/41ac526630da57ad6eb9d02431b17657            venture   

     funded_at  raised_amount_usd                      permalink  \
7   06-08-2014          5000000.0  /organization/all-def-digital   
31  26-02-2015          5000000.0       /organization/chefs-feed   
61  01-08-2006          5000000.0   /organization/huffingtonpost   
62  01-09-2007          5000000.0   /organization/huffingtonpost   
85  01-09-2007         10000000.0        /organization/matchmine   

                   name                           category_list     status  \
7       All Def Digital                                   Media  operating   
31            ChefsFeed     Media|Mobile|Restaurants|Technology  operating   
61  The Huffington Post                   Media|News|Publishing   acquired   
62  The Huffington Post                   Media|News|Publishing   acquired   
85            MatchMine  Media|News|Reviews and Recommendations     closed   

   country_code  raised_amount_million_usd Primary_Sector    main_sector  
7           USA                        5.0          Media  Entertainment  
31          USA                        5.0          Media  Entertainment  
61          USA                        5.0          Media  Entertainment  
62          USA                        5.0          Media  Entertainment  
85          USA                       10.0          Media  Entertainment

In [304]:
# Manual Inspection for merged D1 dataframe
print("Columns : {}".format(D1.columns))
print("\nShape : {}".format(D1.shape))
print("\nVariable types : \n")
D1.info()
print("\nValues : {}".format(D1.values))
D1.describe()

Columns : Index(['funding_round_permalink', 'funding_round_type', 'funded_at',
       'raised_amount_usd', 'permalink', 'name', 'category_list', 'status',
       'country_code', 'raised_amount_million_usd', 'Primary_Sector',
       'main_sector'],
      dtype='object')

Shape : (12092, 12)

Variable types : 

<class 'pandas.core.frame.DataFrame'>
Int64Index: 12092 entries, 7 to 88458
Data columns (total 12 columns):
funding_round_permalink      12092 non-null object
funding_round_type           12092 non-null object
funded_at                    12092 non-null object
raised_amount_usd            12092 non-null float64
permalink                    12092 non-null object
name                         12092 non-null object
category_list                12092 non-null object
status                       12092 non-null object
country_code                 12092 non-null object
raised_amount_million_usd    12092 non-null float64
Primary_Sector               12092 non-null object
main_sector      

raised_amount_usd  raised_amount_million_usd
count       1.209200e+04               12092.000000
mean        8.931698e+06                   8.931700
std         3.081068e+06                   3.081067
min         4.999672e+06                   5.000000
25%         6.000000e+06                   6.000000
50%         8.300000e+06                   8.300000
75%         1.100000e+07                  11.000000
max         1.500010e+07                  15.000000

In [305]:
# creating new dataframe D2 for country United Kingdom (GBR)
# the data is extracted from master_frame that contains
# all the initially required columns
D2 = pd.DataFrame(master_frame[(master_frame.funding_round_type == 'venture') 
                               & (master_frame.country_code == 'GBR') 
                               & (master_frame.raised_amount_million_usd >= 5) 
                               & (master_frame.raised_amount_million_usd <= 15)
                              ]
                 ).copy()
D2.head()

funding_round_permalink funding_round_type  \
1    /funding-round/21a2cbf6f2fb2a1c2a61e04bf930dfe6            venture   
2    /funding-round/bd626ed022f5c66574b1afe234f3c90d            venture   
225  /funding-round/8195587cbd5e51af7514ee92ef4ba6ba            venture   
257  /funding-round/cf3fe3b7c86186b9f478d0ea37613f7a            venture   
258  /funding-round/e5e4ef4ebae63fc36ef0cd57dd20ff1c            venture   

      funded_at  raised_amount_usd                                  permalink  \
1    06-10-2015         15000000.0                        /organization/90min   
2    07-05-2013          5800000.0                        /organization/90min   
225  09-07-2014         10000000.0  /organization/common-interest-communities   
257  01-10-2014          5460000.0                /organization/geospock-ltd-   
258  05-10-2015          5400000.0                /organization/geospock-ltd-   

                            name  \
1                          90min   
2                          90min   
225  Common Interest Communities   
257                GeoSpock Ltd.   
258                GeoSpock Ltd.   

                                        category_list     status country_code  \
1                 Media|News|Publishing|Soccer|Sports  operating          GBR   
2                 Media|News|Publishing|Soccer|Sports  operating          GBR   
225  Application Platforms|Internet|Software|Startups  operating          GBR   
257         Application Platforms|Databases|Real Time  operating          GBR   
258         Application Platforms|Databases|Real Time  operating          GBR   

     raised_amount_million_usd         Primary_Sector  \
1                        15.00                  Media   
2                         5.80                  Media   
225                      10.00  Application Platforms   
257                       5.46  Application Platforms   
258                       5.40  Application Platforms   

                    main_sector  
1                 Entertainment  
2                 Entertainment  
225  News, Search and Messaging  
257  News, Search and Messaging  
258  News, Search and Messaging

In [306]:
# Manual Inspection for merged D2 dataframe
print("Columns : {}".format(D2.columns))
print("\nShape : {}".format(D2.shape))
print("\nVariable types : \n")
D2.info()
print("\nValues : {}".format(D2.values))
D2.describe()

Columns : Index(['funding_round_permalink', 'funding_round_type', 'funded_at',
       'raised_amount_usd', 'permalink', 'name', 'category_list', 'status',
       'country_code', 'raised_amount_million_usd', 'Primary_Sector',
       'main_sector'],
      dtype='object')

Shape : (622, 12)

Variable types : 

<class 'pandas.core.frame.DataFrame'>
Int64Index: 622 entries, 1 to 88266
Data columns (total 12 columns):
funding_round_permalink      622 non-null object
funding_round_type           622 non-null object
funded_at                    622 non-null object
raised_amount_usd            622 non-null float64
permalink                    622 non-null object
name                         622 non-null object
category_list                622 non-null object
status                       622 non-null object
country_code                 622 non-null object
raised_amount_million_usd    622 non-null float64
Primary_Sector               622 non-null object
main_sector                  622 non-null o

raised_amount_usd  raised_amount_million_usd
count       6.220000e+02                 622.000000
mean        8.672152e+06                   8.672151
std         2.876107e+06                   2.876109
min         5.000000e+06                   5.000000
25%         6.088248e+06                   6.088000
50%         8.000000e+06                   8.000000
75%         1.067669e+07                  10.676750
max         1.500000e+07                  15.000000

In [307]:
# creating new dataframe D3 for country India (IND)
# the data is extracted from master_frame that contains
# all the initially required columns
D3 = pd.DataFrame(master_frame[(master_frame.funding_round_type == 'venture') 
                               & (master_frame.country_code == 'IND') 
                               & (master_frame.raised_amount_million_usd >=5) 
                               & (master_frame.raised_amount_million_usd <= 15)
                              ]
                 ).copy()
D3.head()

funding_round_permalink funding_round_type  \
0    /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
550  /funding-round/b72eaac5ea12ac0f50573ac3d6d46b8d            venture   
593  /funding-round/e37673bc7b0f1dfd3782f8f7abdb9ec8            venture   
640  /funding-round/452a7fc1f34df2d3dcda4e28234bc671            venture   
642  /funding-round/f5b252d6442ce231bb01586ca1821f63            venture   

      funded_at  raised_amount_usd                       permalink  \
0    05-01-2015         10000000.0             /organization/-fame   
550  28-04-2015         10000000.0           /organization/babajob   
593  05-02-2008         11750000.0  /organization/bharat-matrimony   
640  24-01-2012          5000000.0     /organization/bluestone-com   
642  18-03-2014         10000000.0     /organization/bluestone-com   

                 name                                      category_list  \
0               #fame                                              Media   
550           Babajob  Curated Web|Information Technology|Services|St...   
593  Bharat Matrimony                           Curated Web|Match-Making   
640     Bluestone.com                                        Curated Web   
642     Bluestone.com                                        Curated Web   

        status country_code  raised_amount_million_usd Primary_Sector  \
0    operating          IND                      10.00          Media   
550  operating          IND                      10.00    Curated Web   
593  operating          IND                      11.75    Curated Web   
640  operating          IND                       5.00    Curated Web   
642  operating          IND                      10.00    Curated Web   

                    main_sector  
0                 Entertainment  
550  News, Search and Messaging  
593  News, Search and Messaging  
640  News, Search and Messaging  
642  News, Search and Messaging

In [308]:
# Manual Inspection for merged D3 dataframe
print("Columns : {}".format(D3.columns))
print("\nShape : {}".format(D3.shape))
print("\nVariable types : \n")
D3.info()
print("\nValues : {}".format(D3.values))
D3.describe()

Columns : Index(['funding_round_permalink', 'funding_round_type', 'funded_at',
       'raised_amount_usd', 'permalink', 'name', 'category_list', 'status',
       'country_code', 'raised_amount_million_usd', 'Primary_Sector',
       'main_sector'],
      dtype='object')

Shape : (328, 12)

Variable types : 

<class 'pandas.core.frame.DataFrame'>
Int64Index: 328 entries, 0 to 88158
Data columns (total 12 columns):
funding_round_permalink      328 non-null object
funding_round_type           328 non-null object
funded_at                    328 non-null object
raised_amount_usd            328 non-null float64
permalink                    328 non-null object
name                         328 non-null object
category_list                328 non-null object
status                       328 non-null object
country_code                 328 non-null object
raised_amount_million_usd    328 non-null float64
Primary_Sector               328 non-null object
main_sector                  328 non-null o

raised_amount_usd  raised_amount_million_usd
count       3.280000e+02                 328.000000
mean        8.992511e+06                   8.992512
std         3.219811e+06                   3.219810
min         5.000000e+06                   5.000000
25%         6.000000e+06                   6.000000
50%         8.700000e+06                   8.700000
75%         1.055500e+07                  10.555000
max         1.500000e+07                  15.000000

In [309]:
# For D1
# extracting the total sum investment per main sector
# putting it in a seperate dataframe
D1_main_sec_sum = pd.DataFrame(D1.groupby('main_sector').sum()['raised_amount_usd'])
D1_main_sec_sum.rename(columns={'raised_amount_usd':'sum_main_sector_usd'},inplace = True)
D1_main_sec_sum.reset_index(inplace=True)
D1_main_sec_sum.head()

main_sector  sum_main_sector_usd
0         Automotive & Sports         1.454104e+09
1  Cleantech / Semiconductors         2.167843e+10
2               Entertainment         5.099198e+09
3                      Health         8.266859e+09
4               Manufacturing         7.298553e+09

In [310]:
# For D1
# extracting the total count for investment per main sector
# putting it in a seperate dataframe
D1_main_sec_count = pd.DataFrame(D1.groupby('main_sector').count()['raised_amount_usd'])
D1_main_sec_count.rename(columns={'raised_amount_usd':'count_main_sector_usd'},inplace = True)
D1_main_sec_count.reset_index(inplace=True)
D1_main_sec_count.head()

main_sector  count_main_sector_usd
0         Automotive & Sports                    167
1  Cleantech / Semiconductors                   2355
2               Entertainment                    591
3                      Health                    914
4               Manufacturing                    803

In [311]:
# merging the aggregated data with dataframe D1
D1 = pd.merge(D1,D1_main_sec_count, how='inner',on = 'main_sector')
D1 = pd.merge(D1,D1_main_sec_sum, how='inner',on = 'main_sector')

D1.head()


funding_round_permalink funding_round_type  \
0  /funding-round/452a2342fe720285c3b92e9bd927d9ba            venture   
1  /funding-round/adca195749ae9ace84684723fbe75e5b            venture   
2  /funding-round/7f05940c4d2dfecb8e50a0e5720e5065            venture   
3  /funding-round/9241ae16e08df17ebdc064e49e23035a            venture   
4  /funding-round/41ac526630da57ad6eb9d02431b17657            venture   

    funded_at  raised_amount_usd                      permalink  \
0  06-08-2014          5000000.0  /organization/all-def-digital   
1  26-02-2015          5000000.0       /organization/chefs-feed   
2  01-08-2006          5000000.0   /organization/huffingtonpost   
3  01-09-2007          5000000.0   /organization/huffingtonpost   
4  01-09-2007         10000000.0        /organization/matchmine   

                  name                           category_list     status  \
0      All Def Digital                                   Media  operating   
1            ChefsFeed     Media|Mobile|Restaurants|Technology  operating   
2  The Huffington Post                   Media|News|Publishing   acquired   
3  The Huffington Post                   Media|News|Publishing   acquired   
4            MatchMine  Media|News|Reviews and Recommendations     closed   

  country_code  raised_amount_million_usd Primary_Sector    main_sector  \
0          USA                        5.0          Media  Entertainment   
1          USA                        5.0          Media  Entertainment   
2          USA                        5.0          Media  Entertainment   
3          USA                        5.0          Media  Entertainment   
4          USA                       10.0          Media  Entertainment   

   count_main_sector_usd  sum_main_sector_usd  
0                    591         5.099198e+09  
1                    591         5.099198e+09  
2                    591         5.099198e+09  
3                    591         5.099198e+09  
4                    591         5.099198e+09

In [312]:
# For D2
# extracting the total sum investment per main sector
# putting it in a seperate dataframe
D2_main_sec_sum = pd.DataFrame(D2.groupby('main_sector').sum()['raised_amount_usd'])
D2_main_sec_sum.rename(columns={'raised_amount_usd':'sum_main_sector_usd'},inplace = True)
D2_main_sec_sum.reset_index(inplace=True)
D2_main_sec_sum.head()

main_sector  sum_main_sector_usd
0         Automotive & Sports         1.670516e+08
1  Cleantech / Semiconductors         1.163990e+09
2               Entertainment         4.827847e+08
3                      Health         2.145375e+08
4               Manufacturing         3.619403e+08

In [313]:
# For D2
# extracting the total count for investment per main sector
# putting it in a seperate dataframe
D2_main_sec_count = pd.DataFrame(D2.groupby('main_sector').count()['raised_amount_usd'])
D2_main_sec_count.rename(columns={'raised_amount_usd':'count_main_sector_usd'},inplace = True)
D2_main_sec_count.reset_index(inplace=True)
D2_main_sec_count.head()

main_sector  count_main_sector_usd
0         Automotive & Sports                     16
1  Cleantech / Semiconductors                    130
2               Entertainment                     56
3                      Health                     24
4               Manufacturing                     42

In [314]:
# merging the aggregated data with dataframe D2
D2 = pd.merge(D2,D2_main_sec_count, how='inner',on = 'main_sector')
D2 = pd.merge(D2,D2_main_sec_sum, how='inner',on = 'main_sector')

D2.head()

funding_round_permalink funding_round_type  \
0  /funding-round/21a2cbf6f2fb2a1c2a61e04bf930dfe6            venture   
1  /funding-round/bd626ed022f5c66574b1afe234f3c90d            venture   
2  /funding-round/d2fc787fbc5e4f468dff8b2c557993f1            venture   
3  /funding-round/47df01ed44d7b5916159051e5e32391e            venture   
4  /funding-round/c6a873b4cbdd7ea3d023a771bd3b2f99            venture   

    funded_at  raised_amount_usd                 permalink        name  \
0  06-10-2015         15000000.0       /organization/90min       90min   
1  07-05-2013          5800000.0       /organization/90min       90min   
2  13-05-2010          8800000.0   /organization/eutechnyx   Eutechnyx   
3  01-06-2011         10000000.0  /organization/mind-candy  Mind Candy   
4  23-11-2006         10860000.0  /organization/mind-candy  Mind Candy   

                         category_list     status country_code  \
0  Media|News|Publishing|Soccer|Sports  operating          GBR   
1  Media|News|Publishing|Soccer|Sports  operating          GBR   
2                                Games  operating          GBR   
3                                Games  operating          GBR   
4                                Games  operating          GBR   

   raised_amount_million_usd Primary_Sector    main_sector  \
0                      15.00          Media  Entertainment   
1                       5.80          Media  Entertainment   
2                       8.80          Games  Entertainment   
3                      10.00          Games  Entertainment   
4                      10.86          Games  Entertainment   

   count_main_sector_usd  sum_main_sector_usd  
0                     56          482784687.0  
1                     56          482784687.0  
2                     56          482784687.0  
3                     56          482784687.0  
4                     56          482784687.0

In [315]:
# For D3
# extracting the total sum investment per main sector
# putting it in a seperate dataframe
D3_main_sec_sum = pd.DataFrame(D3.groupby('main_sector').sum()['raised_amount_usd'])
D3_main_sec_sum.rename(columns={'raised_amount_usd':'sum_main_sector_usd'},inplace = True)
D3_main_sec_sum.reset_index(inplace=True)
D3_main_sec_sum.head()

main_sector  sum_main_sector_usd
0         Automotive & Sports          136900000.0
1  Cleantech / Semiconductors          165380000.0
2               Entertainment          280830000.0
3                      Health          167740000.0
4               Manufacturing          200900000.0

In [316]:
# For D3
# extracting the total count for investment per main sector
# putting it in a seperate dataframe
D3_main_sec_count = pd.DataFrame(D3.groupby('main_sector').count()['raised_amount_usd'])
D3_main_sec_count.rename(columns={'raised_amount_usd':'count_main_sector_usd'},inplace = True)
D3_main_sec_count.reset_index(inplace=True)
D3_main_sec_count.head()

main_sector  count_main_sector_usd
0         Automotive & Sports                     13
1  Cleantech / Semiconductors                     20
2               Entertainment                     33
3                      Health                     19
4               Manufacturing                     21

In [317]:
# merging the aggregated data with dataframe D3
D3 = pd.merge(D3,D3_main_sec_count, how='inner',on = 'main_sector')
D3 = pd.merge(D3,D3_main_sec_sum, how='inner',on = 'main_sector')

D3.head()

funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/6035248811c9530b11bd442d9239a0b1            venture   
2  /funding-round/6b024f4906c288c66d1df966e6aeb256            venture   
3  /funding-round/b095563fd43d1e4fd16da3f4bcd040af            venture   
4  /funding-round/adb94c131e001a7438a4695d873d8dc1            venture   

    funded_at  raised_amount_usd                permalink       name  \
0  05-01-2015         10000000.0      /organization/-fame      #fame   
1  27-11-2006          5000000.0     /organization/dhruva     Dhruva   
2  29-03-2007          5000000.0  /organization/games2win  Games2Win   
3  30-03-2011          6000000.0  /organization/games2win  Games2Win   
4  03-11-2015          5000000.0      /organization/pokkt      POKKT   

  category_list     status country_code  raised_amount_million_usd  \
0         Media  operating          IND                       10.0   
1         Games  operating          IND                        5.0   
2         Games  operating          IND                        5.0   
3         Games  operating          IND                        6.0   
4         Games  operating          IND                        5.0   

  Primary_Sector    main_sector  count_main_sector_usd  sum_main_sector_usd  
0          Media  Entertainment                     33          280830000.0  
1          Games  Entertainment                     33          280830000.0  
2          Games  Entertainment                     33          280830000.0  
3          Games  Entertainment                     33          280830000.0  
4          Games  Entertainment                     33          280830000.0

 ### Extracting the required data to complete the table

- ##### Total number of investments

In [318]:
# Total number of row entries in D1 dataframe is the total number of investments in United States
D1.shape[0]

12092

In [319]:
#Total number of row entries in D2 dataframe is the total number of investments in United Kingdom
D2.shape[0]

622

In [320]:
#Total number of row entries in D3 dataframe is the total number of investments in India
D3.shape[0]

328

- #### Total amount of investment

In [322]:
# Total sum of raised amount in D1 dataframe is the total amount of investments in United States 
D1['raised_amount_usd'].sum()

108002096957.0

In [323]:
# Total sum of raised amount in D2 dataframe is the total amount of investments in United Kingdom
D2['raised_amount_usd'].sum()

5394078692.0

In [324]:
# Total sum of raised amount in D3 dataframe is the total amount of investments in India 
D3['raised_amount_usd'].sum()

2949543602.0

- #### Top 3 main sectors (based on count of investments) and investment in these sectors

In [325]:
# Sum Value across main sectors is already present in dataframe. 
# It is however dupicated across multiple entries.
# To get the top three sectors and their sum values, we have to selet only the 
# columns main_sector and count_main_sector_usd
exp_resultset_D1 = D1[['main_sector','count_main_sector_usd']].drop_duplicates()
    
exp_resultset_D1.sort_values(by='count_main_sector_usd',ascending=False).head(3)

main_sector  count_main_sector_usd
7251                                   Others                   2957
4533  Social, Finance, Analytics, Advertising                   2718
2178               Cleantech / Semiconductors                   2355

In [326]:
# Sum Value across main sectors is already present in dataframe. 
# It is however dupicated across multiple entries.
# To get the top three sectors and their sum values, we have to selet only the 
# columns main_sector and count_main_sector_usd
exp_resultset_D2 = D2[['main_sector','count_main_sector_usd']].drop_duplicates()
    
exp_resultset_D2.sort_values(by='count_main_sector_usd',ascending=False).head(3)

main_sector  count_main_sector_usd
392                                   Others                    148
259  Social, Finance, Analytics, Advertising                    133
129               Cleantech / Semiconductors                    130

In [327]:
# Sum Value across main sectors is already present in dataframe. 
# It is however dupicated across multiple entries.
# To get the top three sectors and their sum values, we have to selet only the 
# columns main_sector and count_main_sector_usd
exp_resultset_D3 = D3[['main_sector','count_main_sector_usd']].drop_duplicates()
    
exp_resultset_D3.sort_values(by='count_main_sector_usd',ascending=False).head(3)

main_sector  count_main_sector_usd
165                                   Others                    110
105  Social, Finance, Analytics, Advertising                     60
33                News, Search and Messaging                     52

- #### Company that received highest investment with `main_sector` as "Others"

In [328]:
topSecCompGrp_D1 = D1[D1.main_sector=='Others'].groupby('name')
topSecCompGrp_D1['raised_amount_usd'].sum().sort_values(ascending=False).head()

name
Virtustream          64300000.0
Capella Photonics    54968051.0
PMW Technologies     54890480.0
AirTight Networks    54201907.0
deCarta              52100000.0
Name: raised_amount_usd, dtype: float64

##### Conclusion : Highest investment company in D1 for main sector "Others" is Virtustream.

In [329]:
topSecCompGrp_D2 = D2[D2.main_sector=='Others'].groupby('name')
topSecCompGrp_D2['raised_amount_million_usd'].sum().sort_values(ascending=False).head()

name
Electric Cloud             37.000
SenSage                    36.250
Enigmatec                  32.500
SilverRail Technologies    29.000
OpenCloud                  27.973
Name: raised_amount_million_usd, dtype: float64

##### Conclusion : Highest investment company in D2 for main sector "Others" is Electric Cloud

In [330]:
topSecCompGrp_D3 = D3[D3.main_sector=='Others'].groupby('name')
topSecCompGrp_D3['raised_amount_million_usd'].sum().sort_values(ascending=False).head()

name
FirstCry.com         39.0
Myntra               38.0
CommonFloor          32.9
Pepperfry.com        28.0
ItzCash Card Ltd.    25.0
Name: raised_amount_million_usd, dtype: float64

##### Conclusion : Highest investment company in D3 for main sector "Others" is FirstCry.com.

In [331]:
top2SecCompGrp_D1 = D1[D1.main_sector=='Social, Finance, Analytics, Advertising'].groupby('name')
top2SecCompGrp_D1['raised_amount_million_usd'].sum().sort_values(ascending=False).head()

name
SST Inc. (Formerly ShotSpotter)    67.933
Demandbase                         63.000
Intacct                            61.800
NetBase Solutions                  60.600
Lotame                             59.700
Name: raised_amount_million_usd, dtype: float64

##### Conclusion : Highest investment company in D1 for main sector 'Social, Finance, Analytics, Advertising' is SST Inc. (Formerly ShotSpotter)

In [332]:
top2SecCompGrp_D2 = D2[D2.main_sector=='Social, Finance, Analytics, Advertising'].groupby('name')
top2SecCompGrp_D2['raised_amount_million_usd'].sum().sort_values(ascending=False).head()

name
Celltick Technologies    37.500
myThings                 34.000
Zopa                     32.900
VisualDNA                28.550
MarketInvoice            25.553
Name: raised_amount_million_usd, dtype: float64

##### Conclusion : Highest investment company in D2 for main sector 'Social, Finance, Analytics, Advertising' is Celltick Technologies.

In [333]:
top2SecCompGrp_D3 = D3[D3.main_sector=='Social, Finance, Analytics, Advertising'].groupby('name')
top2SecCompGrp_D3['raised_amount_million_usd'].sum().sort_values(ascending=False).head()

name
Manthan Systems               50.700
Komli Media                   28.000
ShopClues.com                 25.000
inTarvo                       21.900
Grameen Financial Services    21.556
Name: raised_amount_million_usd, dtype: float64

##### Conclusion : Highest investment company in D3 for main sector 'Social, Finance, Analytics, Advertising' is Manthan Systems.